### DISCUSSION TITLE
"git x --help" not working [Solved]

### DISCUSSION THREAD

**Discussion by gvanem** [git-for-windows/git#3138] [Q&A]
Hello group; first time here.

A long time ago when I did e.g. `git init --help`, I would get the `git-init.html` help-page  launched in my default browser. 
This is not working even after a fresh uninstall and reinstall of `Git-2.31.0-64-bit.exe`.

Instead, Windows' *Explorer* starts wanting me to select the .html-file apparently. How do I enable the
file `f:\ProgramFiler\Git-2\mingw64\share\doc\git-doc\git-init.html` (and others) to be launched directly?
Some Registry setting I'm missing?

**Comment by PhilipOakley** [comment]
It sounds like your installation is slightly non-standard. Completing the issue template would help. In particular it looks like your Git is not at a typical location, and the help files have not been copied to the matching directory.

For me I have:
```
$ git version
git version 2.31.0.windows.1

phili@Philip-Win10 MINGW64 ~
$ git init --help
```
which opens `file:///C:/Program%20Files/Git/mingw64/share/doc/git-doc/git-init.html` nicely laid out

checking my root (`/`) location
```
phili@Philip-Win10 MINGW64 ~
$ cd /

phili@Philip-Win10 MINGW64 /
$ pwd -W
C:/Program Files/Git

phili@Philip-Win10 MINGW64 /
$ which git
/mingw64/bin/git
```

How was Git installed on your machine (is it a personal machine or was it installed by corporate IT, hence why `f:\ProgramFiler\Git-2\`) ?

**Reply by gvanem** [reply]
> git version 2.31.0.windows.1

Same as I have.

> How was Git installed on your machine

Download from `gitforwindows.org`, ran the `Git-2.31.0-64-bit.exe` installer as before. A personal machine. I used `f:\ProgramFiler` since my `c:\` drive if 60% full. Does it really matter for this installer?

I do have `f:\ProgramFiler\Git-2\mingw64\share\doc\git-doc\git-init.html` which is the same relative location you have.

**Reply by gvanem** [reply]
I think I found the problem; I had an `INFOPATH` env-var variable (unrelated to git). 

Removing that or extending it with `;f:\ProgramFiler\Git-2\mingw64\share\doc`,  a `git init --help` works!
But yikes. What does .info-files have to do with the .html-files?

**Reply by PhilipOakley** [reply]
That does sound like a slightly 'buggy' feature. Most of the Git env variable are prefixed with `GIT..` but some env variables are used more generally, so...

Which app is using/providing the `INFOPATH` env-var variable?

Would you be able to help with looking it this? It's open source after all 😃 

The installer is part of the https://github.com/git-for-windows/build-extra/tree/main/installer repo.

**Reply by gvanem** [reply]
> Which app is using

Some old GNU-info program. Can't remember. But lots of hits for `INFOPATH` 
here: https://github.com/git-for-windows/git-sdk-64/search?q=INFOPATH
